test/e2e/mpi_job_test.go

```diff
@@ -230,6 +217,98 @@ var _ = ginkgo.Describe("MPIJob", func() {
 		})
 
 	})
+
+	ginkgo.Context("with scheduler-plugins", func() {
+		var (
+			ctx                    = context.Background()
+			unschedulableResources = &corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("100000"),   // unschedulable
+				corev1.ResourceMemory: resource.MustParse("100000Gi"), // unschedulable
+			}
+		)
+
+		ginkgo.BeforeEach(func() {
+			operator, err := k8sClient.AppsV1().Deployments(mpiOperator).Get(ctx, mpiOperator, metav1.GetOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			operator.Spec.Template.Spec.Containers[0].Args = append(operator.Spec.Template.Spec.Containers[0].Args, "--gang-scheduling=scheduler-plugins-scheduler")
+			_, err = k8sClient.AppsV1().Deployments(mpiOperator).Update(ctx, operator, metav1.UpdateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			gomega.Eventually(func() bool {
+				updatedDeployment, err := k8sClient.AppsV1().Deployments(mpiOperator).Get(ctx, mpiOperator, metav1.GetOptions{})
+				return updatedDeployment.Status.AvailableReplicas != int32(0) && err == nil
+			}, foreverTimeout, waitInterval).Should(gomega.BeTrue())
+			createMPIJobWithOpenMPI(mpiJob)
+			mpiJob.Spec.RunPolicy.SchedulingPolicy = &kubeflow.SchedulingPolicy{MinResources: unschedulableResources}
+			// We should wait for informer caches to sync.
```

Comment:
I faced the below error without timout:

```shell
STEP: Waiting for Pods to created
• Failure [0.566 seconds]
MPIJob
/Users/tenzen-y/go/src/github.com/kubeflow/mpi-operator/test/e2e/mpi_job_test.go:35
  with scheduler-plugins
  /Users/tenzen-y/go/src/github.com/kubeflow/mpi-operator/test/e2e/mpi_job_test.go:216
    should create pending pods [It]
    /Users/tenzen-y/go/src/github.com/kubeflow/mpi-operator/test/e2e/mpi_job_test.go:262

    Unexpected error:
        <*errors.StatusError | 0x140003dd400>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "podgroups.scheduling.x-k8s.io \"pi\" not found",
                Reason: "NotFound",
                Details: {
                    Name: "pi",
                    Group: "scheduling.x-k8s.io",
                    Kind: "podgroups",
                    UID: "",
                    Causes: nil,
                    RetryAfterSeconds: 0,
                },
                Code: 404,
            },
        }
        podgroups.scheduling.x-k8s.io "pi" not found
    occurred

    /Users/tenzen-y/go/src/github.com/kubeflow/mpi-operator/test/e2e/mpi_job_test.go:291
```

I guess the old controller (without the `--gang-scheduling` flag) created Launcher and Worker since pods don't have `shceulerName: scheduler-plugins-scheduler` field and the podGroup resource doesn't exist in the cluster.

So, we may need to verify whether the old controller was completely deleted instead of using timeout.

Reply:
Through more investigation, I realized that I could fix the above error by just setting strategy=Recreate to mpi-operator Deployment instead of waiting to be deleted an old pod.

Reply:
Done.